src/heightmap.cpp

```diff
@@ -409,44 +410,52 @@ void FixSlopes()
 	/* Top and left edge */
 	for (row = 0; (uint)row < height; row++) {
 		for (col = 0; (uint)col < width; col++) {
-			current_tile = MAX_TILE_HEIGHT;
+			current_height = MAX_TILE_HEIGHT;
 			if (col != 0) {
 				/* Find lowest tile; either the top or left one */
-				current_tile = TileHeight(TileXY(col - 1, row)); // top edge
+				current_height = TileHeight(TileXY(col - 1, row)); // top edge
 			}
 			if (row != 0) {
-				if (TileHeight(TileXY(col, row - 1)) < current_tile) {
-					current_tile = TileHeight(TileXY(col, row - 1)); // left edge
+				if (TileHeight(TileXY(col, row - 1)) < current_height) {
+					current_height = TileHeight(TileXY(col, row - 1)); // left edge
 				}
 			}
 
 			/* Does the height differ more than one? */
-			if (TileHeight(TileXY(col, row)) >= (uint)current_tile + 2) {
+			TileIndex tile = TileXY(col, row);
+			if (TileHeight(tile) >= (uint)current_height + 2) {
 				/* Then change the height to be no more than one */
-				SetTileHeight(TileXY(col, row), current_tile + 1);
+				SetTileHeight(tile, current_height + 1);
+				if (IsInnerTile(tile) && RandomRange(max_height) <= current_height) {
```

Comment:
Could use a comment to explain why RandomRange() is used here, as well as noting that instead of cliffs, we get rocks.

Reply:
Hopefully makes sense now?